src/main/java/leelawatcher/goboard/MarkablePosition.java

```diff
@@ -153,67 +146,33 @@ private boolean isOnBoard(PointOfPlay p, int boardSize) {
    * @param boardSize The size of the board
    * @return A HashSet of PointOfPlay objects describing the group
    */
-
-  public HashSet getGroupSet(PointOfPlay p, HashSet<PointOfPlay> members, int boardSize) {
+  public Set getGroupSet(PointOfPlay p, Set<PointOfPlay> members, int boardSize) {
     if (members == null) {
       members = new HashSet<>();
       clearMarks();                // NOTE that this clears all marks
-
     }
+
     if (isOnBoard(p, boardSize)) {
       setMark(p);
 
       if (stoneAt(p)) {
-        PointOfPlay dir;
         members.add(p);
-        dir = new PointOfPlay(p.getX(), p.getY() + 1);
-        if (isOnBoard(dir, boardSize) && !isMarked(dir)
-            && ((blackAt(p) && blackAt(dir))
-            || (whiteAt(p) && whiteAt(dir)))) {
-          getGroupSet(dir, members, boardSize);
-        }
-        dir = new PointOfPlay(p.getX() + 1, p.getY());
-        if (isOnBoard(dir, boardSize) && !isMarked(dir)
-            && ((blackAt(p) && blackAt(dir))
-            || (whiteAt(p) && whiteAt(dir)))) {
-          getGroupSet(dir, members, boardSize);
-        }
-        dir = new PointOfPlay(p.getX(), p.getY() - 1);
-        if (isOnBoard(dir, boardSize) && !isMarked(dir)
-            && ((blackAt(p) && blackAt(dir))
-            || (whiteAt(p) && whiteAt(dir)))) {
-          getGroupSet(dir, members, boardSize);
-        }
-        dir = new PointOfPlay(p.getX() - 1, p.getY());
-        if (isOnBoard(dir, boardSize) && !isMarked(dir)
-            && ((blackAt(p) && blackAt(dir))
-            || (whiteAt(p) && whiteAt(dir)))) {
-          getGroupSet(dir, members, boardSize);
-        }
-
+        checkNbr(p, p.getX(), p.getY() + 1, members, boardSize);
+        checkNbr(p, p.getX() + 1, p.getY(), members, boardSize);
+        checkNbr(p, p.getX(), p.getY() - 1, members, boardSize);
+        checkNbr(p, p.getX() - 1, p.getY(), members, boardSize);
       }
     }
     return members;
   }
 
-}
+  private void checkNbr(PointOfPlay p, int x, int y, Set<PointOfPlay> members, int boardSize) {
```

Comment:
Please write out whatever Nbr stands for. I like long method names, my fingers will survive the typing but my brain might not survive the confusion... :)